src/model.rs

```diff
@@ -337,15 +340,23 @@ impl Model<PluginsIncluded> {
         self.scip
             .create_prob(name)
             .expect("Failed to create problem in state PluginsIncluded");
-        self.move_to_state::<ProblemCreated>()
+        Model {
+            scip: mem::take(&mut self.scip),
+            state: ProblemCreated {
+                vars: BTreeMap::new(),
+                conss: Vec::new(),
+            },
+        }
     }
 
     pub fn read_prob(&mut self, filename: &str) -> Result<Model<ProblemCreated>, Retcode> {
         self.scip.read_prob(filename)?;
-        self.vars = self.scip.get_vars();
-        self.conss = self.scip.get_conss();
-        let new_model = self.move_to_state::<ProblemCreated>();
-
+        let vars = self.scip.get_vars();
+        let conss = self.scip.get_conss();
+        let new_model = Model {
+            scip: mem::take(&mut self.scip),
```

Comment:
same here

Reply:
And also the variables and constraints I guess.